lighthouse-core/lib/asset-saver.js

```diff
@@ -54,19 +56,93 @@ img {
   `;
 }
 
+const artifactsFilename = 'artifacts.json';
+const traceSuffix = '.trace.json';
+const devtoolsLogSuffix = '.devtoolslog.json';
+
 /**
- * Save entire artifacts object to a single stringified file located at
- * pathWithBasename + .artifacts.log
+ * Load artifacts object from files located within basePath
+ * Also save the traces to their own files
+ * @param {string} basePath
+ * @return {!Promise<!Artifacts>}
+ */
+// Set to ignore because testing it would imply testing fs, which isn't strictly necessary.
+/* istanbul ignore next */
+function loadArtifacts(basePath) {
+  log.log('Reading artifacts from disk:', basePath);
+  // load artifacts.json
+  const filenames = fs.readdirSync(basePath);
+  const artifacts = JSON.parse(fs.readFileSync(path.join(basePath, artifactsFilename), 'utf8'));
+
+  // load devtoolsLogs
+  artifacts.devtoolsLogs = {};
+  filenames.filter(f => f.endsWith(devtoolsLogSuffix)).map(filename => {
+    const passName = filename.replace(devtoolsLogSuffix, '');
+    const devtoolsLog = JSON.parse(fs.readFileSync(path.join(basePath, filename), 'utf8'));
+    artifacts.devtoolsLogs[passName] = devtoolsLog;
+  });
+
+  // load traces
+  artifacts.traces = {};
+  const promises = filenames.filter(f => f.endsWith(traceSuffix)).map(filename => {
+    return new Promise(resolve => {
+      const passName = filename.replace(traceSuffix, '');
+      const readStream = fs.createReadStream(path.join(basePath, filename), {
+        encoding: 'utf-8',
+        highWaterMark: 4 * 1024 * 1024, // TODO benchmark to find the best buffer size here
+      });
+      const parser = new TraceParser();
+      readStream.on('data', chunk => parser.parseChunk(chunk));
+      readStream.on('end', _ => {
+        artifacts.traces[passName] = parser.getTrace();
+        resolve();
+      });
+    });
+  });
+  return Promise.all(promises).then(_ => artifacts);
+}
+
+/**
+ * Save artifacts object mostly to single file located at basePath/artifacts.log.
+ * Also save the traces & devtoolsLogs to their own files
  * @param {!Artifacts} artifacts
- * @param {string} pathWithBasename
+ * @param {string} basePath
  */
 // Set to ignore because testing it would imply testing fs, which isn't strictly necessary.
 /* istanbul ignore next */
-function saveArtifacts(artifacts, pathWithBasename) {
-  const fullPath = `${pathWithBasename}.artifacts.log`;
-  // The networkRecords artifacts have circular references
-  fs.writeFileSync(fullPath, stringifySafe(artifacts));
-  log.log('artifacts file saved to disk', fullPath);
+function saveArtifacts(artifacts, basePath) {
+  if (!fs.existsSync(basePath)) {
+    fs.mkdirSync(basePath);
+  }
+  rimraf.sync(`${basePath}/*${traceSuffix}`);
```

Comment:
I believe chrome launcher had a lot of issues on windows when using sync rimraf. Should we move to async?

Reply:
rimraf's bumps since then were around this issue so i am hoping we're good now. https://github.com/isaacs/rimraf/commits/master